error.o builds without -fPIC

Building on Ubuntu 20.04, cmake 3.22, gcc 10.3, no `-fPIC` for `error.o`:
```
[ 42%] Building C object include/CMakeFiles/postgresql_h3_shared.dir/error.c.o
cd /home/user/h3-pg/build/include && /usr/bin/cc  -isystem /usr/include/postgresql -isystem /usr/include/postgresql/14/server -O3 -DNDEBUG -MD -MT include/CMakeFiles/postgresql_h3_shared.dir/error.c.o -MF CMakeFiles/postgresql_h3_shared.dir/error.c.o.d -o CMakeFiles/postgresql_h3_shared.dir/error.c.o -c /home/user/h3-pg/include/error.c
```

The build fails on Rocky Linux 8 ([Dockerfile.txt](https://github.com/zachasme/h3-pg/files/10754673/Dockerfile.txt)) with a linking error:
```
[ 80%] Linking C shared module h3.so
/usr/bin/ld: ../include/CMakeFiles/postgresql_h3_shared.dir/error.c.o: relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: Nonrepresentable section on output
collect2: error: ld returned 1 exit status
```
The reason the build works on Ubuntu is probably that gcc is built with `--enable-default-pie` on Ubuntu (and some other distributions), but not on Rocky.

--- a/include/CMakeLists.txt
+++ b/include/CMakeLists.txt
@@ -7,3 +7,4 @@ target_link_libraries(postgresql_h3_shared
 target_include_directories(postgresql_h3_shared
   INTERFACE ./
 )
+set_target_properties(postgresql_h3_shared PROPERTIES POSITION_INDEPENDENT_CODE True)

